sdk/eventhub/azure-eventhub/samples/async_samples/authenticate_with_azure_sas_credential_async.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+"""
+Example to demonstrate utilizing SAS (Shared Access Signature) tokens to authenticate with ServiceBus
+"""
+
+# pylint: disable=C0111
+
+import asyncio
+import os
+import time
+import hmac
+import hashlib
+import base64
+try:
+    from urllib.parse import quote as url_parse_quote
+except ImportError:
+    from urllib import pathname2url as url_parse_quote
+
+from azure.core.credentials import AccessToken, AzureSasCredential
+from azure.eventhub.aio import EventHubProducerClient
+from azure.eventhub import EventData
+
+
+def generate_sas_token(uri, sas_name, sas_value, token_ttl):
+    """Performs the signing and encoding needed to generate a sas token from a sas key."""
+    sas = sas_value.encode('utf-8')
+    expiry = str(int(time.time() + token_ttl))
+    string_to_sign = (uri + '\n' + expiry).encode('utf-8')
+    signed_hmac_sha256 = hmac.HMAC(sas, string_to_sign, hashlib.sha256)
+    signature = url_parse_quote(base64.b64encode(signed_hmac_sha256.digest()))
+    return 'SharedAccessSignature sr={}&sig={}&se={}&skn={}'.format(uri, signature, expiry, sas_name)
+
+# Target namespace and hub must also be specified.  Consumer group is set to default unless required otherwise.
+FULLY_QUALIFIED_NAMESPACE = os.environ['EVENT_HUB_HOSTNAME']
+EVENTHUB_NAME = os.environ['EVENT_HUB_NAME']
+CONSUMER_GROUP = "$Default"
+
+# The following part creates a SAS token. Users can use any way to create a SAS token.
```

Comment:
maybe specify exactly what these ways are so users knows? Or, if there are a lot, mention a few. Something like this?
`# There are multiple ways to create a SAS token for authentication, such as *insert a couple of ways*. The following is one example.` (Same for async sample.)